LICENSE

```diff
@@ -1,2 +1,3 @@
-Unless specifically indicated otherwise in a file, files are licensed
-under the Apache 2.0 license, as can be found in: apache-2.0.txt
+Unless specifically indicated otherwise in a file, files are dual licensed
+under either the Apache 2.0 license OR the GNU General Public License version 2
+or later, at your option, as can be found in: apache-2.0.txt and gpl-2.0.txt
```

Comment:
It might be just me, but I feel a slight ambiguity in the placement of "at your option". We clearly mean that the options are Apache 2.0 or GPL 2.0 or later, but it might be misinterpreted and assumed that the term refers to GPL 2.0 or later versions. I think it is clear from the context that this is not the case, but usually even the slightest ambiguity makes me very uncomfortable when reading a document like this.

I was wondering if something like this would be clearer:

```suggestion
under your choice of the Apache 2.0 license OR the GNU General Public License version 2
or later, as can be found in: apache-2.0.txt and gpl-2.0.txt
```

What do you think?

Reply:
Of course if this is some standard text for such disjunctive dual licences, then we probably shouldn't change it...

Reply:
“At your option” is commonplace in contracts (not just licenses). But that doesn't mean we have to use it.

I agree that the sentence is ambiguous. Has it been validated by a legal professional? As a non-legal professional, I would split the sentence in two and rephrase it a little.

> Unless specifically indicated otherwise in a file, files are licensed, at your option, under either the Apache License version 2.0, or the GNU General Public License version 2 or later. The Apache License version 2.0 can be found in apache-2.0.txt and the GNU General Public License version 2 can be found in gpl-2.0.txt.

Reply:
Can we use the language already in the License section of the readme and SPDX? For example, 

"Unless specifically indicated otherwise in a file, Mbed TLS files are provided under Apache License 2.0, or GNU General Public License v2.0 or later (SPDX-License-Identifier: Apache-2.0 OR GPL-2.0-or-later). A copy of these licenses can be found in apache-2.0.txt and gpl-2.0.txt."

At the same time, I think we should update the License section in the readme to:
"Unless specifically indicated otherwise in a file, Mbed TLS files are provided under the [Apache-2.0](https://spdx.org/licenses/Apache-2.0.html) OR [GPL-2.0-or-later](https://spdx.org/licenses/GPL-2.0-or-later.html) licenses. A copy of these licenses can be found in [apache-2.0.txt](apache-2.0.txt) and [gpl-2.0.txt](gpl-2.0.txt). Contributors must accept that their contributions are made under both the Apache-2.0 AND GPL-2.0-or-later licenses."

Note I've dropped the last sentence which is no longer needed in the LTS branches.

Finally the commit message should avoid the term "dual licensing" and refer to precise SPDX identifiers.

Reply:
@danh-arm Done.